ts/features/wallet/onboarding/bancomat/screens/search/SearchBankComponent.tsx

```diff
@@ -0,0 +1,147 @@
+import { Content, Input, Item, Text, View } from "native-base";
+import * as React from "react";
+import {
+  SafeAreaView,
+  FlatList,
+  ListRenderItemInfo,
+  ActivityIndicator
+} from "react-native";
+import { debounce } from "lodash";
+import I18n from "../../../../../../i18n";
+import { Body } from "../../../../../../components/core/typography/Body";
+import { Link } from "../../../../../../components/core/typography/Link";
+import { Label } from "../../../../../../components/core/typography/Label";
+import IconFont from "../../../../../../components/ui/IconFont";
+import FooterWithButtons from "../../../../../../components/ui/FooterWithButtons";
+import { Abi } from "../../../../../../../definitions/pagopa/bancomat/Abi";
+import { sortAbiByName } from "../../utils/abi";
+import { BankPreviewItem } from "../../components/BankPreviewItem";
+import {
+  cancelButtonProps,
+  confirmButtonProps
+} from "../../../../../bonus/bonusVacanze/components/buttons/ButtonConfigurations";
+import { IOColors } from "../../../../../../components/core/variables/IOColors";
+
+type Props = {
+  bankList: ReadonlyArray<Abi>;
+  isLoading: boolean;
+  onItemPress: (abi: string) => void;
+  openTosModal: () => void;
+  onCancel: () => void;
+  onContinue: () => void;
+};
+
+export const SearchBankComponent: React.FunctionComponent<Props> = (
+  props: Props
+) => {
+  const [searchText, setSearchText] = React.useState("");
+  const [filteredList, setFilteredList] = React.useState<ReadonlyArray<Abi>>(
+    []
+  );
+  const [isSearchStarted, setIsSearchStarted] = React.useState(false);
+
+  const performSearch = (text: string) => {
```

Comment:
It seems it isn't working.
This is because `performSearch` is used within `React.useRef` so its scope never changes.
In particular `props.bankList` has the first value (an empty array) when the component mounts so the search produces always an empty result.

```suggestion
  const performSearch = (text: string, bankList: ReadonlyArray<Abi>) => {
```
then add `props.bankList` in useEffect

`debounceRef.current(searchText,props.bankList);`